servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -356,11 +349,13 @@ private HttpExecutionContext buildExecutionContext(final HttpExecutionStrategy s
                 filteredService, drainRequestPayloadBody);
     }
 
-    private HttpExecutionStrategy computeServiceStrategy(Object service) {
-        HttpExecutionStrategy serviceStrategy = requiredOffloads(service, defaultStrategy());
+    private HttpExecutionStrategy computeServiceStrategy(HttpExecutionStrategyInfluencer service) {
+        HttpExecutionStrategy serviceStrategy = service.requiredOffloads();
         HttpExecutionStrategy filterStrategy = computeRequiredStrategy(serviceFilters, serviceStrategy);
-        return defaultStrategy() == strategy ? filterStrategy :
-                strategy.hasOffloads() ? strategy.merge(filterStrategy) : offloadNone();
+        return defaultStrategy() == strategy ?
+            filterStrategy.hasOffloads() ? strategy.merge(filterStrategy) : strategy :
+            // We currently treat the builder strategy specially, allowing it to override the filter strategy.
+            strategy.hasOffloads() ? strategy : offloadNone();
```

Comment:
In this case, looks like by some reasons users passed a custom strategy to the builder that does not have offloading. As a user, I would expect to see my custom strategy on the `ServerContext`. `offloadNone()` will be surprising to me. Maybe I have a special subtype that carries more information/state.

Reply:
The strategy that is exposed is in the `ExecutionContext` currently almost always `OFFLOAD_EVENT_CLOSE_STRATEGY`  because there is always a non-offloading filter (`ClearAsyncContext`) and the real offloading is done by the `OffloadingFilter`.

Reply:
I mean that users can implement their own `HttpExecutionStrategy`, keep `hasOffloads()` as `false`, but have some other flags or state. Don't think replacing it with `offloadNone()` is reasonable. I would expect this to be:
```java
return defaultStrategy() == strategy ?
            (filterStrategy.hasOffloads() ? filterStrategy : strategy) :
            (filterStrategy.hasOffloads() ? strategy.merge(filterStrategy) : strategy);
```

Reply:
Same but in different way:
```java
return filterStrategy.hasOffloads() ?
            (defaultStrategy() == strategy ? filterStrategy : strategy.merge(filterStrategy)) : strategy;
```

Reply:
no longer applicable